### ISSUE TITLE
Simply inheriting from collections.abc.Callable causes reportGeneralTypeIssues

### ISSUE THREAD

**Issue by Achxy** [microsoft/pyright#3854]
**Describe the bug**
Inheriting from the `collections.abc.Callable` class causes an `Argument to class must be a base class (reportGeneralTypeIssues)` error.

**To Reproduce**
```python
from collections.abc import Callable

class Foo(Callable):  # Argument to class must be a base class Pyright (reportGeneralTypeIssues)
    pass
```

**Expected behavior**
No errors are emitted, and inheritance takes place as normal causing the child class to have `__call__` as an abstract method

**Medium**
VSCode extension, version: `v1.1.267`

**Comment by erictraut** [comment]
`Callable` is a special form, not a normal class, so I wouldn't expect this to work. I confirmed that it doesn't cause a runtime exception, but I think it's legitimate for a type checker to flag this as an incorrect usage. Mypy also generates an error in this case.

**Comment by davetapley** [comment]
Linking these together for others, since it also caught me out:
- https://github.com/python/mypy/issues/14392
